aten/src/ATen/core/ivalue_inl.h

```diff
@@ -1222,6 +1228,13 @@ inline IValue::IValue(c10::intrusive_ptr<at::Quantizer> v)
   payload.u.as_intrusive_ptr = null_to_undefined_tensor(v.release());
 }
 
+template <typename T>
+inline IValue::IValue(c10::complex<T> c)
+    : tag(Tag::ComplexDouble), is_intrusive_ptr(true) {
+  auto v = c10::make_intrusive<ivalue::ComplexHolder>(c);
+  payload.u.as_intrusive_ptr = v.release();
```

Comment:
Anjali appears to be copying how other intrusive pointers are implemented. This implementation look correct to me: IValue takes ownership of the ComplexHolder, and then is responsible for destructing it at the end. What is subtle here (and not commented yet, see above) is that we have to indirect complex numbers through intrusive pointer because they are too big to fit in IValue